option.go

```diff
@@ -40,3 +41,10 @@ func WithReadFallbackToRawString() Option {
 		opt.readFallbackToRawString = true
 	}
 }
+
+// This sets a separator for queryStr in HCLEditor funcs. The default separator is ".".
+func WithQuerySeparator(chr string) Option {
```

Comment:
I think we should limit the subset of characters here. We should not allow for any string to be used.

At the very least, please add validation that the provided string is of length 1.

Ideally we also only allow a subset of non-word characters as separators (e.g. `/`, `.`, `|`, etc.), but that's not a blocker.

Reply:
Or, change this to be `rune` so this is handled by the compiler.
```suggestion
func WithQuerySeparator(chr rune) Option {
```

Reply:
used `rune`. 

> At the very least, please add validation that the provided string is of length 1.

It is hard to tell because string in go is just a sequence of bytes. `rune` instead makes sure of a single character in the sense of unicode.